backend/src/main/java/de/jmpsoftware/backend/service/ArtNetThread.java

```diff
@@ -0,0 +1,123 @@
+package de.jmpsoftware.backend.service;
+
+import ch.bildspur.artnet.ArtNetClient;
+import de.jmpsoftware.backend.model.PipeEntry;
+import lombok.SneakyThrows;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.time.ZonedDateTime;
+
+public class ArtNetThread extends Thread {
+    private static final Log LOG = LogFactory.getLog(ArtNetThread.class);
+    public static final int ARTNET_FORCE_BROADCAST_TIME = 500;
+    public static final int ACTIVE_ARTNET_DOZY_TIME = 50;
+    public static final int SLEEPING_ARTNET_DOZY_TIME = 1000;
+
+
+    final ArtNetClient artnet;
+    final byte[][] dmxData = new byte[ArtNetService.COUNT_UNIVERSE][ArtNetService.SIZE_UNIVERSE];
+    private final PipedInputStream pipe;
+    final PipeEntry pipeEntry;
+    final byte[] readByte = new byte[4];
+    long sendDMXTimeStamp;
+
+    public ArtNetThread(PipedInputStream pipe) {
+        this.pipe = pipe;
+        this.artnet = new ArtNetClient(null);
+        pipeEntry = new PipeEntry();
+        for (int i = 0; i < ArtNetService.COUNT_UNIVERSE; i++)
+            for (int ii = 0; ii < ArtNetService.SIZE_UNIVERSE; ii++)
+                dmxData[i][ii] = 0;
+
+        LOG.trace("ArtNetClient is created");
+    }
+
+    @SneakyThrows
+    @Override
+    public void run() {
+        LOG.trace("ArtNetClient Thread goes run");
+        this.artnet.start();
+        sendDMXTimeStamp = ZonedDateTime.now().toInstant().toEpochMilli();
+
+        while (true) {
+            while (pipe.available() > 0) {
+                if (checkPipeReadDoneCorrect(pipe.read(readByte, 0, 4))) {
```

Comment:
I am a little bit concerned about the nesting, maybe this can be refactored to be a little bit simpler (good thing you have tests 👍)